segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StorageEventProcessor.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.segmentstore.server.containers;
+
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.common.util.BufferView;
+import io.pravega.segmentstore.server.ContainerEventProcessor;
+import io.pravega.segmentstore.storage.chunklayer.AbstractTaskQueueManager;
+import io.pravega.segmentstore.storage.chunklayer.GarbageCollector;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Implementation of {@link AbstractTaskQueueManager} that uses {@link io.pravega.segmentstore.server.ContainerEventProcessor.EventProcessor}
+ * as underlying implementation.
+ * This class acts as adaptor that converts calls on {@link io.pravega.segmentstore.server.ContainerEventProcessor.EventProcessor}
+ * into appropriate calls on {@link AbstractTaskQueueManager} and vice versa.
+ */
+@Slf4j
+public class StorageEventProcessor implements AbstractTaskQueueManager<GarbageCollector.TaskInfo> {
+
+    private final static GarbageCollector.TaskInfo.Serializer SERIALIZER = new GarbageCollector.TaskInfo.Serializer();
+
+    private final int containerID;
+    private final ContainerEventProcessor eventProcessor;
+    private final Function<List<GarbageCollector.TaskInfo>, CompletableFuture<Void>> callBack;
+    private final int maxItemsAtOnce;
+    private final String traceObjectId;
+    @Getter
+    private final ConcurrentHashMap<String, ContainerEventProcessor.EventProcessor> eventProcessorMap = new ConcurrentHashMap<>();
+
+    /**
+     * Constructor.
+     * @param containerID Container id.
```

Comment:
Nit: keep a blank line between the comment and the subsequent lines with tags (like `@param`). Normally you keep that blank line, but there are cases like this one and others in which there is no blank line.